src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+
+/**
+ * @ingroup UCC_LIB
+ * @brief Modify the configuration descriptor
+ *
+ * @param [in] config   Pointer to the configuration descriptor to be modified
+ * @param [in] name     Configuration variable to be modified
+ * @param [in] value    Configuration value to set
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_lib_config_modify(ucc_lib_config_t *config, const char *name,
+                                    const char *value);
+
+
+/**
+ * @ingroup UCC_LIB
+ *
+ * @brief A local operation to initialize and allocate the resources for the UCC
+ * operations. The parameters passed using the ucc_lib_params_t and
+ * ucc_lib_config structures will customize and select the functionality of the
+ * UCC library. The library can be customized for its interaction with the user
+ * threads, types of collective operations, and reductions supported.
+ * On success, the library object will be created and ucc_status_t will return
+ * UCC_OK. On error, the library object will not be created and corresponding
+ * error code as defined by ucc_status_t is returned.
+
+ * @param [in]  params    user provided parameters to customize the library functionality
+ * @param [in]  config    UCC configuration descriptor allocated through
+ *                        @ref ucc_config_read "ucc_config_read()" routine.
+ * @param [out] lib_p     UCC library handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ *
+ */
+
+ucc_status_t ucc_init(const ucc_lib_params_t *params,
+                      const ucc_lib_config_t *config,
+                      ucc_lib_h *lib_p);
+
+/**
+ * @ingroup UCC_LIB
+ * @brief @ref ucc_finalize is a local operation to release the resources and
+ * cleanup. All participants that invoked @ref ucc_init should call this
+ * routine.
+ *
+ * @param [in] lib_p   Handle to @ref ucc_lib_h
+ *                     "UCC library".
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_finalize(ucc_lib_h lib_p);
+
+
+/**
+ * @ingroup UCC_LIB
+ * @brief A query operation to get the attributes of the library object. The
+ * attributes are library configured values and reflect the choices made by the
+ * library implementation.
+ *
+ * @param [out]  lib_atrib  Library attributes
+ * @param [in]   lib_p      Input library object
+ * 
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_lib_get_attribs(ucc_lib_h lib_p, ucc_lib_attribs_t *lib_atrib);
+
+/**
+ * *************************************************************
+ *                   Context Section 
+ * *************************************************************
+ */
+
+
+
+typedef enum {
+   UCC_CONTEXT_EXCLUSIVE = 0 ,
+   UCC_CONTEXT_SHARED
+} ucc_context_type_t;
+
+enum ucc_context_attribs_field {
+    UCC_CONTEXT_ATTRIBS_TYPE                   = UCC_BIT(0),
+    UCC_CONTEXT_ATTRIBS_COLL_SYNC_TYPE         = UCC_BIT(1),
+    UCC_CONTEXT_ATTRIBS_COLL_OOB               = UCC_BIT(2)
```

Comment:
COLL_OOB is probably should not be part of attribs? It is always passed by user.

Reply:
Agree.